streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListener.scala

```diff
@@ -39,6 +39,17 @@ case class StreamingListenerBatchCompleted(batchInfo: BatchInfo) extends Streami
 case class StreamingListenerBatchStarted(batchInfo: BatchInfo) extends StreamingListenerEvent
 
 @DeveloperApi
+case class StreamingListenerOutputOperationStarted(
+    outputOperationId: Int,
+    outputOperationCallSite: String,
+    startTime: Long) extends StreamingListenerEvent
+
+@DeveloperApi
+case class StreamingListenerOutputOperationCompleted(
```

Comment:
I think its better for future API compatibility to have another class called `OutputOperationInfo` (similar to `BatchInfo`). We can add more field in `OutputOperationInfo` later, without breaking binary compatibility.